collect_app/src/main/java/org/odk/collect/android/formmanagement/ServerFormDownloader.java

```diff
@@ -292,6 +296,20 @@ private static void moveMediaFiles(String tempMediaPath, File formMediaPath) thr
         }
     }
 
+    private void copySavedFileFromPreviousFormVersionIfExists(String formId, String mediaDirPath) {
```

Comment:
I think it'd be good to extract this logic to a use case (probably a new Kotlin object with a method like `FormEntryUseCases`) so you can write unit tests around the details here. For instance, right now you can change the `max` to a `min` and no tests fail.

Reply:
Done.